src/extract.jl

```diff
@@ -154,12 +154,15 @@ function git_file_hash(
     # TODO: this largely duplicates the logic of read_data
     # read_data could be used directly if SHA offered an interface
     # where you write data to an IO object and it maintains a hash
+    t = round_up(size)
     while size > 0
-        r = readbytes!(tar, buf, size < sizeof(buf) ? round_up(size) : sizeof(buf))
-        r < 512 && eof(io) && error("premature end of tar file")
+        n = min(t, length(buf))
+        r = readbytes!(tar, buf, n)
+        r < n && eof(io) && error("premature end of tar file")
         v = view(buf, 1:min(r, size))
         SHA.update!(ctx, v)
         size -= length(v)
+        t -= r
```

Comment:
If `sizeof(buf)` is not a multiple of 512, this will usually make `t` a non-multiple of 512 and it will no longer be aligned with `round_up(size)`.

Reply:
Never mind, I guess that's fine.